code/datums/helper_datums/teleport.dm

```diff
@@ -39,22 +39,50 @@
 		return 1
 	return 0
 
+/datum/teleport/proc/checkInhibitors(atom/adestination)
+	if(istype(adestination))
+		var/list/turf/good_turfs = list()
+		var/list/turf/bad_turfs = list()
+		for(var/found_inhibitor in circlerange(adestination,8))
+			if(!istype(found_inhibitor,/obj/machinery/anti_bluespace))
+				continue
+			var/obj/machinery/anti_bluespace/AB = found_inhibitor
+			if(AB.stat & (NOPOWER | BROKEN) )
+				continue
+			bad_turfs += circlerangeturfs(get_turf(AB),8)
+			good_turfs += circlerangeturfs(get_turf(AB),9)
+		if(good_turfs.len && bad_turfs.len)
+			good_turfs -= bad_turfs

```

Comment:
@skull132 tell me if this is expensive, I cannot tell. good_turfs is of 251 tiles and bad_turfs is 201 tiles. Is it worth writing proc that gets just the perimeter tiles? Or making it work on map AREA?

Reply:
I Guess it should be fine since it is not called often